### PR TITLE
Top down biomarker III

### PR DESCRIPTION
**Pull request by trishorts**
Previously, C-terminal truncation products did not allow for variable initiator methionine. This PR fixes that bug.

I removed the capacity to deal w/ Retain/Cleave/Variable in initiator methionine behaviour. The default is variable methionine for everything. This simplified the code

The word biomarker has been substituted throughout by the word truncation

### CODE REVIEW

File: mzLib/Proteomics/Protein/Protein.cs [hunk 1]
```diff
@@ -741,80 +703,25 @@ public void AddBiomarkers(bool addFullProtein = true, bool addForEachOrigninalPr
...
             CleaveOnceBetweenProteolysisProducts();
         }
 
-        /// <summary>
-        /// When a protein has existing proteolysis products, we have to remove methionine when appropriate before creating additional proteolysis products
-        /// </summary>
-        /// <param name="existingProducts"></param>
-        /// <param name="initiatorMethionineBehavior"></param>
-        public void RemoveMethionineWhenAppropriateFromExistingProduts(InitiatorMethionineBehavior initiatorMethionineBehavior)
+        public void AddIntactProteoformToProteolysisProducts(int minProductBaseSequenceLength)
```
Comment (acesnik) [inline mzLib/Proteomics/Protein/Protein.cs:720]: Could you add a method summary?

File: mzLib/Proteomics/Protein/Protein.cs [hunk 2]
```diff
@@ -587,91 +587,54 @@ public void RestoreUnfilteredModifications()
...
+                if (addNterminalDigestionTruncations)
                 {
-                    AddNterminalBiomarkers(lengthOfProteolysis, fullProteinOneBasedBegin, fullProteinOneBasedEnd, minProductBaseSequenceLength, proteolyisisProductName);
+                    AddNterminalTruncations(lengthOfProteolysis, fullProteinOneBasedBegin, fullProteinOneBasedEnd, minProductBaseSequenceLength, proteolyisisProductName);
                 }
             }
         }
 
-        private void AddCterminalBiomarkers(int lengthOfProteolysis, int fullProteinOneBasedEnd, int fullProteinOneBasedBegin, int minProductBaseSequenceLength, string proteolyisisProductName)
+        private void AddCterminalTruncations(int lengthOfProteolysis, int fullProteinOneBasedEnd, int fullProteinOneBasedBegin, int minProductBaseSequenceLength, string proteolyisisProductName)
```
Comment (acesnik) [inline mzLib/Proteomics/Protein/Protein.cs:637]: Could you add a method summary?

File: mzLib/Proteomics/Protein/Protein.cs [hunk 3]
```diff
@@ -684,7 +647,7 @@ private void AddCterminalBiomarkers(int lengthOfProteolysis, int fullProteinOneB
             }
         }
 
-        private void AddNterminalBiomarkers(int lengthOfProteolysis, int fullProteinOneBasedBegin, int fullProteinOneBasedEnd, int minProductBaseSequenceLength, string proteolyisisProductName)
+        private void AddNterminalTruncations(int lengthOfProteolysis, int fullProteinOneBasedBegin, int fullProteinOneBasedEnd, int minProductBaseSequenceLength, string proteolyisisProductName)
```
Comment (acesnik) [inline mzLib/Proteomics/Protein/Protein.cs:650]: Could you add a method summary?

File: mzLib/Proteomics/Protein/Protein.cs [hunk 4]
```diff
@@ -587,91 +587,54 @@ public void RestoreUnfilteredModifications()
             return validModDictionary;
         }
 
-        public void AddBiomarkersToProteolysisProducts(int fullProteinOneBasedBegin, int fullProteinOneBasedEnd, bool addNterminalDigestionBiomarkers, bool addCterminalDigestionBiomarkers, InitiatorMethionineBehavior initiatorMethionineBehavior, int minProductBaseSequenceLength, int lengthOfProteolysis, string proteolyisisProductName)
+        public void AddTruncationsToProteolysisProducts(int fullProteinOneBasedBegin, int fullProteinOneBasedEnd, bool addNterminalDigestionTruncations, bool addCterminalDigestionTruncations, int minProductBaseSequenceLength, int lengthOfProteolysis, string proteolyisisProductName)
```
Comment (acesnik) [inline mzLib/Proteomics/Protein/Protein.cs:590]: Could you add a method summary?

**Review by acesnik — APPROVED** [review]
Looks good as far as I can tell. 👍🏼  Some method summaries might help clarify the code a bit if you want to add those.

File: mzLib/Proteomics/Protein/Protein.cs (hunk 1 above)
Comment (trishorts) [reply mzLib/Proteomics/Protein/Protein.cs]: done

File: mzLib/Proteomics/Protein/Protein.cs (hunk 4 above)
Comment (trishorts) [reply mzLib/Proteomics/Protein/Protein.cs]: done

File: mzLib/Proteomics/Protein/Protein.cs (hunk 3 above)
Comment (trishorts) [reply mzLib/Proteomics/Protein/Protein.cs:659]: done

**Review by acesnik — APPROVED** [review]

File: mzLib/Proteomics/Protein/Protein.cs [hunk 5]
```diff
@@ -706,29 +678,28 @@ private void AddNterminalBiomarkers(int lengthOfProteolysis, int fullProteinOneB
         /// </summary>
         /// <param name="addFullProtein"> This needs to be added to the proteolysisProducts list to be searched </param>
         /// <param name="addForEachOrigninalProteolysisProduct"> the original products are there but those resulting from N- or C-terminal degradation still need to be added</param>
-        /// <param name="addNterminalDigestionBiomarkers"></param>
-        /// <param name="addCterminalDigestionBiomarkers"></param>
+        /// <param name="addNterminalDigestionTruncations"></param>
+        /// <param name="addCterminalDigestionTruncations"></param>
         /// <param name="initiatorMethionineBehavior"> this effects the intact proteoform as well as any original proteolysis products containing the N-terminus</param>
```
Comment (nbollis) [inline mzLib/Proteomics/Protein/Protein.cs:683]: initiatorMethionineBehavior is no longer a parameter

Comment (trishorts) [reply mzLib/Proteomics/Protein/Protein.cs:683]: good catch!

Comment (trishorts) [reply mzLib/Proteomics/Protein/Protein.cs]: fixed

**Review by nbollis — APPROVED** [review]
10/10 would review again

**Review by acesnik — APPROVED** [review]

**Pull request merged**